src/requirements/requirement-frontend-utils.ts

```diff
@@ -501,3 +503,49 @@ export function getRelatedUnfulfilledRequirements(
   }
   return { directlyRelatedRequirements, selfCheckRequirements };
 }
+
+export function getAllEligibleRequirements(
```

Comment:
This function contains conditionals & loops that are fairly deeply nested--I think it might be more readable to factor some of this code into separate functions

Reply:
Don't think refactors can help a lot. These logic requires to be colocated together...